tests_metricflow/snapshots/test_conversion_metric_rendering.py/SqlQueryPlan/DuckDB/test_conversion_metric_with_categorical_filter__plan0_optimized.sql

```diff
@@ -18,20 +18,28 @@ FROM (
       , visit__referrer_id
       , SUM(visits) AS visits
     FROM (
-      -- Read Elements From Semantic Model 'visits_source'
-      -- Metric Time Dimension 'ds'
+      -- Constrain Output with WHERE
       -- Pass Only Elements: ['visits', 'visit__referrer_id', 'metric_time__day']
       SELECT
-        DATE_TRUNC('day', ds) AS metric_time__day
-        , referrer_id AS visit__referrer_id
-        , 1 AS visits
-      FROM ***************************.fct_visits visits_source_src_28000
-    ) subq_18
+        metric_time__day
+        , visit__referrer_id
+        , visits
+      FROM (
+        -- Read Elements From Semantic Model 'visits_source'
+        -- Metric Time Dimension 'ds'
+        SELECT
+          DATE_TRUNC('day', ds) AS metric_time__day
+          , referrer_id AS visit__referrer_id
+          , 1 AS visits
+        FROM ***************************.fct_visits visits_source_src_28000
+      ) subq_17
+      WHERE visit__referrer_id = 'ref_id_01'
```

Comment:
Note for reviewers - this is the one irreconcilable difference between the existing build-time pushdown and the optimizer approach. For conversion metrics the build-time pushdown is able short-circuit any pushdown operations for query-level filters, because we can see that the metric type is conversion in advance of our pushdown application.

With the optimizer, the first indication we get that we are dealing with a conversion metric is at the conversion metric join node. We short-circuit all where constraint pushdown at that stage, ensuring that we don't push past the conversion metric computation boundary, but we may push predicates down to the outer edge of the conversion metric join node itself. This is functionally identical to the original logic, but it adds this extra where constraint node at the moment.

The additional subquery should vanish when we update the optimizer to remove filters applied via pushdown.